2020/02/24/symfony-quickstart.html.md

```diff
@@ -0,0 +1,429 @@
+
+As we could see earlier, Symfony actions are returning a Response. One can send a JSON as Response, or even a JSONResponse, as the [documentation](https://symfony.com/doc/current/components/http_foundation.html#creating-a-json-response) describes. One can even specify a JSONP callback via
+
+```php
+$response->setCallback('handleResponse')
+```
+
+Don’t be afraid of this, it’s not very difficult and it’s well-documented.
+
+Requests are also handled [nicely](https://symfony.com/doc/current/introduction/http_fundamentals.html),
+
+```php
+$request->query->get('id')
+```
+
+gets a  GET parameter, while
+
+```php
+$request->request->get('category', 'default value')
+```
+
+is getting a post value.
+
+```php
+$request->query->all()
+```
+
+retrieves all GET parameters
+
+```php
+$request->request->all()
+```
+
+retrieves all POST parameters.
+
+## Caching
+
+Symfony supports caching with the caching module, which, if switched on will cache configuration and routes. If controller actions or routes are changed, then it is advisable to clear the cache via
+
+```
+bin/console cache:clear && bin/console cache:warmup
+```
+
+Whether or not to switch on caching for your application at dev mode depends on the frequency of a need to cache cleanup. Read more [here].
+
+## Configuration
+
+Symfony configuration can be done via YAML, XML or PHP files. In the projects I have been working with, configuration was done in YAML files and they were stored in the config folder inside the root folder of the project.
+
+Inside the **src** folder we can find **Kernel.php** which glues together the project.
+
+## Twig
+
+Symfony is well-paired with Twig, a popular template engine. Why don’t we use PHP for this purpose, one might ask. The answer is that we can use PHP or Smarty or whatever we intend to use for view templating, nobody forces us to use Twig.
+
+Using template engines instead of allowing the developers to use PHP as a template engine has the drawback of limiting the options, however, the exact reason they are used, especially by large teams is the exact same thing that we can outline as its drawback. Yes, it limits the options of the developers, but this way developers will not find it so easy and convenient to temporarily (?) implement their business logic as part of the view.
+
+If my opinion matters, I’m okay with using PHP as a template engine as long as we do not mix up the layers of the application and, by convention the view will remain “only” the view. I’m also okay with using template engines, even though it is less convenient to use when we need to do some experimenting and writing some code for a very short time, after which it is reverted. In that kind of experimental work one feels that enforcing this separation is unnecessary bureaucracy, but the gain is that we will not have to deal with views where lots of things are implemented which should not be there. So, this is a harmless form of bureaucracy, if such a thing exists.
```

Comment:
Remove "If my opinion matters," -- this is your blog post, and your opinion matters!

I think you nicely weighed the pros and cons of PHP vs. other templating languages and your opinion makes perfect sense here.